snapshot/snapshotter.go

```diff
@@ -34,13 +37,14 @@ type Snapshotter interface {
 }
 
 func FromContainerdSnapshotter(name string, s snapshots.Snapshotter, idmap *idtools.IdentityMapping) Snapshotter {
-	return &fromContainerd{name: name, Snapshotter: s, idmap: idmap}
+	return &fromContainerd{name: name, Snapshotter: s, idmap: idmap, redirectDirOption: determineRedirectDirOption()}
 }
 
 type fromContainerd struct {
 	name string
 	snapshots.Snapshotter
-	idmap *idtools.IdentityMapping
+	idmap             *idtools.IdentityMapping
+	redirectDirOption string
```

Comment:
I don't quite the need for this property. Why not just put a public method for this is overlay package, with `sync.Once` ensuring proc read happens only once. And the places (2 I think) that need to overwrite the mounts call into these functions.

Reply:
Thank you for the review. Fixed to avoid adding the property.